tests/integration/strategy_tests.cpp

```diff
@@ -276,6 +276,92 @@ TYPED_TEST(StrategyTest, getActualSize)
   this->m_allocator->deallocate(data);
 }
 
+template <typename T>
+class ReleaseTest : public ::testing::Test {
+ public:
+  void SetUp() override
+  {
+    auto& rm = umpire::ResourceManager::getInstance();
+    std::string name{"release_test_" + std::to_string(unique_strategy_id++)};
+    std::string limiter_name{"limiter_" + std::to_string(unique_strategy_id++)};
+    
+    m_limiter_allocator = 
+        new umpire::Allocator(rm.makeAllocator<umpire::strategy::SizeLimiter>(
+        limiter_name, rm.getAllocator("HOST"), max_alloc_size * num_allocs + padding));
+
+    m_allocator = 
+        new umpire::Allocator(rm.makeAllocator<T>(name, rm.getAllocator(limiter_name), 
+        max_alloc_size * num_allocs));
+  }
+
+  void TearDown() override
+  {
+    delete m_allocator;
+    m_allocator = nullptr;
+
+    delete m_limiter_allocator;
+    m_limiter_allocator = nullptr;
+  }
+
+  ////////////////////////////////////////////////////////////////////////
+  //max_alloc_size: Biggest allocation size used in test.
+  //num_allocs: The number of total allocations 
+  //padding: Some strategies have built-in alignment handling which will
+  //  interfere will this test. Padding helps make sure we account for that.
+  ////////////////////////////////////////////////////////////////////////
+  static const int max_alloc_size = 1024;
+  static const int num_allocs = 8;
+  static const int padding = 64;
+
+  umpire::Allocator* m_allocator;
+  umpire::Allocator* m_limiter_allocator;
+
+};
+
+template <>
+void ReleaseTest<umpire::strategy::FixedPool>::SetUp()
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+  std::string name{"release_test_" + std::to_string(unique_strategy_id++)};
+  std::string limiter_name{"limiter_" + std::to_string(unique_strategy_id++)};
+  
+  static const int max_alloc_size = 1024;
+  static const int num_allocs = 8;
+
+  m_limiter_allocator = 
+        new umpire::Allocator(rm.makeAllocator<umpire::strategy::SizeLimiter>(
+        limiter_name, rm.getAllocator("HOST"), max_alloc_size * num_allocs));
+  
+  m_allocator =
+      new umpire::Allocator(rm.makeAllocator<umpire::strategy::FixedPool>(
+          name, rm.getAllocator(limiter_name), max_alloc_size, 1));
+}
+
+using ReleaseStrategies = ::testing::Types<
+  umpire::strategy::DynamicPoolList,
+  umpire::strategy::DynamicPoolMap, umpire::strategy::FixedPool, 
+  umpire::strategy::QuickPool>;
+
+TYPED_TEST_SUITE(ReleaseTest, ReleaseStrategies, );
+
+TYPED_TEST(ReleaseTest, ReleaseCheck)
+{
+  void* test[this->num_allocs];
+
+  for (int i = 0; i < this->num_allocs; i++) {
+    test[i] = this->m_allocator->allocate(this->max_alloc_size);
+  }
```

Comment:
I think it would be good to call `release` before the `deallocate` loop (which shouldn't do anything). This would have caught the issue I commented on above.